### PR TITLE
HOTFIX: resolve overload ambiguity

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by bbejeck** [comment]
as a side note, while not as concise as a method ref  but `s -> s.toUpperCase()` works too and might look better than a cast, but not that important IMHO

**Comment by mjsax** [comment]
Using `(v) -> v.toUpperCase` gives a warning telling you can pass in a method reference (ie, what the original code does). Thus, the cast seems to be preferable.

File: src/main/java/io/confluent/examples/streams/MapFunctionLambdaExample.java [hunk 1]
```diff
@@ -120,7 +121,7 @@ public static void main(final String[] args) throws Exception {
     final KStream<byte[], String> textLines = builder.stream("TextLinesTopic", Consumed.with(byteArraySerde, stringSerde));
 
     // Variant 1: using `mapValues`
-    final KStream<byte[], String> uppercasedWithMapValues = textLines.mapValues(String::toUpperCase);
+    final KStream<byte[], String> uppercasedWithMapValues = textLines.mapValues((ValueMapper<String, String>) String::toLowerCase);
```
Comment (dguy) [inline src/main/java/io/confluent/examples/streams/MapFunctionLambdaExample.java]: out of curiosity what is the issue?

Comment (mjsax) [reply src/main/java/io/confluent/examples/streams/MapFunctionLambdaExample.java]: This:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:testCompile (default-testCompile) on project kafka-streams-examples: Compilation failure: Compilation failure:
[ERROR] /home/****/workspace/afka-streams-examples_PR-55-ZE4XP7RR5GNQKLRZHHKVNNWF2RAXJEI44CF2FVKHT2EHEL5ECZQA/src/test/java/io/confluent/examples/streams/FanoutLambdaIntegrationTest.java:[96,46] reference to mapValues is ambiguous
[ERROR] both method <VR>mapValues(org.apache.kafka.streams.kstream.ValueMapper<? super V,? extends VR>) in org.apache.kafka.streams.kstream.KStream and method <VR>mapValues(org.apache.kafka.streams.kstream.ValueMapperWithKey<? super K,? super V,? extends VR>) in org.apache.kafka.streams.kstream.KStream match
```

The added overload `mapValue(KeyValueMapper)` cannot be distinguished to the existing `mapValue(ValueMapper)` -- it's a lambda issue.

Comment (dguy) [reply src/main/java/io/confluent/examples/streams/MapFunctionLambdaExample.java]: oic. thanks

**Pull request merged**